plugins/destination/duckdb/client/migrate.go

```diff
@@ -29,118 +28,95 @@ type tableInfo struct {
 	columns []columnInfo
 }
 
-func (c *Client) duckdbTables(tables schema.Schemas) (schema.Schemas, error) {
-	var schemaTables schema.Schemas
+func (c *Client) duckdbTables(tables schema.Tables) (schema.Tables, error) {
+	var schemaTables schema.Tables
 	for _, table := range tables {
-		tableName := schema.TableName(table)
-		info, err := c.getTableInfo(tableName)
+		info, err := c.getTableInfo(table.Name)
 		if err != nil {
 			return nil, err
 		}
 		if info == nil {
 			continue
 		}
-		fields := make([]arrow.Field, len(info.columns))
+		columns := make(schema.ColumnList, len(info.columns))
 		for i, col := range info.columns {
-			md := make(map[string]string)
-			if col.pk {
-				md[schema.MetadataPrimaryKey] = schema.MetadataTrue
-			} else {
-				md[schema.MetadataPrimaryKey] = schema.MetadataFalse
-			}
-			if col.unique {
-				md[schema.MetadataUnique] = schema.MetadataTrue
-			} else {
-				md[schema.MetadataUnique] = schema.MetadataFalse
-			}
-			fields[i] = arrow.Field{
-				Name:     col.name,
-				Type:     c.duckdbTypeToSchema(col.typ),
-				Nullable: !col.notNull,
-				Metadata: arrow.MetadataFrom(md),
+			columns[i] = schema.Column{
+				Name:       col.name,
+				Type:       c.duckdbTypeToSchema(col.typ),
+				NotNull:    col.notNull,
+				PrimaryKey: col.pk,
+				Unique:     col.unique,
 			}
 		}
-		md := arrow.MetadataFrom(map[string]string{
-			schema.MetadataTableName: tableName,
+		schemaTables = append(schemaTables, &schema.Table{
+			Name:    table.Name,
+			Columns: columns,
 		})
-		schemaTables = append(schemaTables, arrow.NewSchema(fields, &md))
 	}
 
 	return schemaTables, nil
 }
 
-func (c *Client) normalizeColumns(tables schema.Schemas) schema.Schemas {
-	var normalized schema.Schemas
+func (c *Client) normalizeColumns(tables schema.Tables) schema.Tables {
+	var normalized schema.Tables
 	for _, table := range tables {
-		fields := table.Fields()
-		for i := range fields {
+		normalizedTable := *table
+		normalizedTable.Columns = make(schema.ColumnList, len(table.Columns))
+		for i := range table.Columns {
 			// In DuckDB, a PK column must be NOT NULL, so we need to make sure that the schema we're comparing to has the same
 			// constraint.
-			metadata := fields[i].Metadata.ToMap()
-
+			normalizedColumn := table.Columns[i]
 			if !c.enabledPks() {
-				metadata[schema.MetadataPrimaryKey] = schema.MetadataFalse
-				metadata[schema.MetadataUnique] = schema.MetadataFalse
-			} else {
-				if schema.IsPk(fields[i]) {
-					fields[i].Nullable = false
-				} else {
-					metadata[schema.MetadataPrimaryKey] = schema.MetadataFalse
-				}
-				if !schema.IsUnique(fields[i]) {
-					metadata[schema.MetadataUnique] = schema.MetadataFalse
-				}
+				normalizedColumn.PrimaryKey = false
+				normalizedColumn.Unique = false
+			} else if normalizedColumn.PrimaryKey {
+				normalizedColumn.NotNull = true
```

Comment:
```suggestion
			} else {
				normalizedColumn.NotNull = normalizedColumn.NotNull || normalizedColumn.PrimaryKey
```

Reply:
What's the different? does it produce a different logic ?

Reply:
just eliminates branching here